src/checkpoint.ts

```diff
@@ -15,41 +16,38 @@ import { AsyncMySqlPool, createMySqlPool } from './mysql';
 import { createPgPool } from './pg';
 import { checkpointConfigSchema } from './schemas';
 import { register } from './register';
-import {
-  ContractSourceConfig,
-  CheckpointConfig,
-  CheckpointOptions,
-  CheckpointWriters,
-  TemplateSource
-} from './types';
-
-const BLOCK_PRELOAD = 1000;
+import { ContractSourceConfig, CheckpointConfig, CheckpointOptions, TemplateSource } from './types';
+
+const BLOCK_PRELOAD_START_RANGE = 1000;
+const BLOCK_RELOAD_MIN_RANGE = 10;
+const BLOCK_PRELOAD_STEP = 100;
+const BLOCK_PRELOAD_TARGET = 10;
 const BLOCK_PRELOAD_OFFSET = 50;
 const DEFAULT_FETCH_INTERVAL = 2000;
 
 export default class Checkpoint {
   public config: CheckpointConfig;
-  public writer: CheckpointWriters;
   public opts?: CheckpointOptions;
   public schema: string;
 
   private readonly entityController: GqlEntityController;
   private readonly log: Logger;
-  private readonly networkProvider: BaseProvider;
+  private readonly indexer: EvmIndexer;
```

Comment:
Any reason to have something EVM specific on this file?

Reply:
Good catch, it should be `BaseIndexer`.